api/analyzers/csharp/analyzer.py

```
@@ -0,0 +1,137 @@
+        if node.type in ['class_declaration', 'interface_declaration', 'enum_declaration',
+                         'struct_declaration', 'method_declaration', 'constructor_declaration']:
+            return node.child_by_field_name('name').text.decode('utf-8')
+        raise ValueError(f"Unknown entity type: {node.type}")
+
+    def get_entity_docstring(self, node: Node) -> Optional[str]:
+        if node.type in ['class_declaration', 'interface_declaration', 'enum_declaration',
+                         'struct_declaration', 'method_declaration', 'constructor_declaration']:
+            if node.prev_sibling and node.prev_sibling.type == "comment":
+                return node.prev_sibling.text.decode('utf-8')
+            return None
+        raise ValueError(f"Unknown entity type: {node.type}")
```

Ruff (0.15.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)